[docs/objects] Update "Use Sui Move Library" (#6048)
MystenLabs/sui: doc/src/build/move/sui-move-library.md

- Exclusively owned by an address.
- Exclusively owned by another object.
- Shared and immutable.

> - Shared and mutable.

Reviewer: See comment below but we probably want to make this 
```
- Immutable.
- Shared.
```

This call will fully consume the object, making it no longer accessible in the current transaction.
Once an address owns an object, for any future use (either read or write) of this object, the signer of the transaction must be the owner of the object.

### Owned by another object

> An object can be owned by another object when the former is added as a [dynamic object field](../programming-with-objects/ch5-dynamic-fields.md) of the latter. Note that the ownership is only tracked in Sui. From Move's perspective, these two objects are still more or less independent in that the child object isn't stored as part of the parent object, but they are associated with each other:

Reviewer: > From Move's perspective, these two objects are still more or less independent in that the child object isn't stored as part of the parent object, but they are associated with each other:

This comment confuses me a bit. From Move's perspective, I would say the object is just wrapped. There is no "ownership" like there was with the child object API. 
Thoughts?

Author: I'll try and generally steer away from the "ownership" terminology.

Author: I also want to distinguish this state from being wrapped because we're talking about dynamic object fields and not dynamic fields here.

```
use sui::dynamic_object_field as ofield;

let a: &A = /* ... */;

> let b: &B = /* ... */;

Reviewer: ```suggestion
let b: B = /* ... */;
```

// Adds `b` as a dynamic object field to `a` with "name" `0: u8`.
ofield::add<u8, B>(&mut a.id, 0, b);
```

> Once an object is owned by another object, it is required that any references to that object go through its owner: If an object-owned object is passed as an input to an entry function in a transaction, that transaction will fail. For instance, if we have a chain of ownership: address `Addr1` owns object `a`, object `a` owns object `b`, and `b` owns object `c` then in order to use object `c` in a Move call, the transaction must be signed by `Addr1`, and accept `a` as an input, and `b` and `c` must be accessed dynamically during transaction execution:

Reviewer: ```suggestion
Once an object is contained as a dynamic field, it can only be accessed through its owner through the `dynamic_object_field` APIs. If an object-owned object is passed as an input to an entry function in a transaction, that transaction will fail. For instance, if we have a chain of ownership: address `Addr1` owns object `a`, object `a` owns object `b`, and `b` owns object `c` then in order to use object `c` in a Move call, the transaction must be signed by `Addr1`, and accept `a` as an input, and `b` and `c` must be accessed dynamically during transaction execution:
```

```
use sui::dynamic_object_field as ofield;

// signed of ctx is Addr1
public entry fun entry_function(a: &A, ctx: &mut TxContext) {
  let b: &B = ofield::borrow<u8, B>(&a.id, 0);
  let c: &C = ofield::borrow<u8, C>(&b.id, 0);
}
```

More examples of how objects can be transferred and owned can be found in
[object_owner.move](https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/unit_tests/data/object_owner/sources/object_owner.move).

> ### Shared and immutable

Reviewer: ```suggestion
### Immutable
```

Reviewer: Some real old terminology here, and we might want to fix some things below. We used to refer to immutable objects as shared immutable and shared objects as shared mutable. 
But we now just focus on the important access patterns for those, so "immutable" and "shared" objects.

To make an object `obj` shared and immutable, one can call:
```
transfer::freeze_object(obj);
```
After this call, `obj` becomes immutable which means it can never be mutated or deleted. This process is also irreversible: once an object is frozen, it will stay frozen forever. An immutable object can be used as reference by anyone in their Move call.

> ### Shared and mutable

Reviewer: ```suggestion
### Shared
```
